python/ray/air/util/data_batch_conversion.py

```diff
@@ -109,6 +110,47 @@ def convert_pandas_to_batch_type(
         )
 
 
+def _convert_batch_type_to_numpy(
+    data: DataBatchType,
+) -> Union[np.ndarray, Dict[str, np.ndarray]]:
+    """Convert the provided data to a NumPy ndarray or dict of ndarrays.
+
+    Args:
+        data: Data of type DataBatchType
+
+    Returns:
+        A numpy representation of the input data.
+    """
+    if isinstance(data, np.ndarray):
+        return data
+    elif isinstance(data, dict):
+        if data.keys() == {TENSOR_COLUMN_NAME} and isinstance(
+            data[TENSOR_COLUMN_NAME], np.ndarray
+        ):
```

Comment:
Hmm I don't know if this should ever happen? This function is/will be used downstream of (1) inline preprocessing during online prediction and (2) streaming ingest preprocessing.
1. Shouldn't be needed since this tensor column scheme is a Datasets artifact that shouldn't make it to DL predictors in Ray Serve or batch non-DL prediction (the latter has a Pandas narrow waist).
2. Shouldn't be needed since it currently is tied to a Pandas narrow waist, and even when it's ported to not be an out-of-band map_batches() call (so it's actually going through the AIR batch format logic, using `Preprocessor.transform()`, etc.), it will be using the `"numpy"` batch format for `.map_batches()` in this case, so dict of ndarrays with this tensor column should never slip through.

Have you encountered a case where this is needed?

Reply:
I think this part is legacy conditional written while i thought you meant we convert all single tensor column format to tensor, but that should be dataset pyarrow internal. Let me remove it